contracts/base/SignatureHandler.sol

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+/// ─────────────────────────────────────────────────────────────────────────────
+///     _   __    _  __
+///    / | / /__ | |/ /_  _______
+///   /  |/ / _ \|   / / / / ___/
+///  / /|  /  __/   / /_/ (__  )
+/// /_/ |_/\___/_/|_\__,_/____/
+///
+/// ─────────────────────────────────────────────────────────────────────────────
+/// Nexus: A suite of contracts for Modular Smart Account compliant with ERC-7579
+/// and ERC-4337, developed by Biconomy. Learn more at https://biconomy.io.
+/// To report security issues, please contact us at: security@biconomy.io
+
+import { EIP712 } from "solady/src/utils/EIP712.sol";
+
+import { Storage } from "./Storage.sol";
+
+/// @title SignatureHandler
+/// @notice Handles EIP-712 signature validation for the Nexus suite.
+/// @dev Implements nested EIP-712 signature validation to prevent replays.
+contract SignatureHandler is EIP712, Storage {
```

Comment:
I am okay with moving utils to SignatureHandler. but not EIP712
prefer Nexus to be is EIP712

Reply:
Most functions here use `EIP712`. Moving them to nexus.sol would make compilation impossible, except for `_domainNameAndVersion`. We can revert to the previous version if you think that it should remains on Nexus. I was just annoyed by the huge detailed natspec for `_erc1271HashForIsValidSignatureViaNestedEIP712` function which might be reducing its visibility.

Reply:
I know it's ideal and was thinking of moving some code too. but for now just leave it be until late stage of audit? we can revist